precompiles/xtokens/src/lib.rs

```diff
@@ -0,0 +1,245 @@
+// Copyright 2019-2021 PureStake Inc.
+// This file is part of Moonbeam.
+
+// Moonbeam is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Moonbeam is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Moonbeam.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Precompile to xtokens runtime methods via the EVM
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use evm::{executor::PrecompileOutput, Context, ExitError, ExitSucceed};
+use frame_support::dispatch::{Dispatchable, GetDispatchInfo, PostDispatchInfo};
+use pallet_evm::{AddressMapping, Precompile};
+use precompile_utils::{
+	error, Address, Bytes, EvmData, EvmDataReader, EvmDataWriter, EvmResult, Gasometer,
+	RuntimeHelper,
+};
+
+use sp_core::{H160, U256};
+use sp_std::{
+	convert::{TryFrom, TryInto},
+	fmt::Debug,
+	marker::PhantomData,
+	vec::Vec,
+};
+mod encoding;
+use encoding::Encoder;
+use sp_std::boxed::Box;
+use xcm::v1::{AssetId, Fungibility, Junctions, MultiAsset, MultiLocation};
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+
+pub type BalanceOf<Runtime> = <Runtime as orml_xtokens::Config>::Balance;
```

Comment:
Typically `BalanceOf` refers to the native MOVR balance type. It isn't wrong to do it this way, but I worry it could be confusing in the future given that other Precompiles care about the MOVR balance. Maybe this one could be called `XBalanceOf`?

Reply:
Perfect